front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -602,63 +602,179 @@ const handleUpdateTimetableItem = async ({
+    updatedPacedTrain = {
       ...basePacedTrain,
-      exceptions: isPacedTrainResponseWithPacedTrainId(timetableItem)
-        ? checkChangeGroups(basePacedTrain, timetableItem.exceptions)
+      exceptions: isPacedTrainResponseWithPacedTrainId(timetableItemToUpdate)
+        ? checkChangeGroups(basePacedTrain, timetableItemToUpdate.exceptions)
         : [],
     };
     updatedTimetableItem = await storePacedTrain(
-      timetableItemId,
+      timetableItemToUpdate.id,
       updatedPacedTrain,
       timetableId,
       dispatch,
       addUpsertedTimetableItems,
       addDeletedTimetableItemIds
     );
   }
-  // TODO: handle roundtrips
-  state.timetableItemIdByNgeId.set(trainrun.id, [updatedTimetableItem.id, null]);
-};
 
-const handleDeleteTimetableItem = async (
-  trainrunId: number,
-  state: MacroEditorState,
-  dispatch: AppDispatch,
-  addDeletedTimetableItemIds: (timetableItemIds: TimetableItemId[]) => void
-) => {
-  // TODO: handle roundtrips
-  const timetableItemId = state.timetableItemIdByNgeId.get(trainrunId)![0];
-  const editoastTrainId = isPacedTrainId(timetableItemId)
-    ? extractEditoastIdFromPacedTrainId(timetableItemId)
-    : extractEditoastIdFromTrainScheduleId(timetableItemId);
-  const endpoint = isPacedTrainId(timetableItemId)
-    ? osrdEditoastApi.endpoints.deletePacedTrain
-    : osrdEditoastApi.endpoints.deleteTrainSchedule;
-  await dispatch(
-    endpoint.initiate({
-      body: { ids: [editoastTrainId] },
-    })
-  ).unwrap();
-  addDeletedTimetableItemIds([timetableItemId]);
-  state.timetableItemIdByNgeId.delete(trainrunId);
+  if (trainrun.direction === 'one_way') {
+    if (timetableIds[1]) {
+      // NGE always selects the forward trip by default when going from round trip to one way trip,
+      // thus the trip that needs to be deleted is always the return trip
+      if (isPacedTrainId(timetableItemToUpdate.id))
+        await dispatch(
+          osrdEditoastApi.endpoints.postRoundTripsPacedTrains.initiate({
+            roundTrips: {
+              one_ways: [extractEditoastIdFromTimetableItemId(timetableItemToUpdate.id)],
+            },
+          })
+        );
+      else
+        await dispatch(
+          osrdEditoastApi.endpoints.postRoundTripsTrainSchedules.initiate({
+            roundTrips: {
+              one_ways: [extractEditoastIdFromTimetableItemId(timetableItemToUpdate.id)],
+            },
+          })
+        );
+      await deleteTimetableItemById(timetableIds[1], dispatch, addDeletedTimetableItemIds);
+    }
+
+    state.timetableItemIdByNgeId.set(trainrun.id, [updatedTimetableItem.id, null]);
+    return;
+  }
+
+  const { path: returnPath, schedule: returnSchedule } = generatePathAndSchedule(
+    trainrunSections,
+    netzgrafikDto.nodes,
+    startDate,
+    TRAINRUN_DIRECTIONS.BACKWARD
+  );
+
+  let newReturnTimetableItem: TimetableItem;
+
+  if (timetableIds[1]) {
+    // update return if already present
+    if (paced) {
```

Comment:
Nit: we could change this check to `if (updatedPacedTrain)` so that we don't need to use the `!` operator below.